### ISSUE TITLE
Cannot switch Apple IDs (and Apple restricting count of free certificates)

### ISSUE THREAD

**Issue by amandamana** [Matchstic/ReProvision#107]
The new experimental1-3 versions have an apparent issue with more than one active certificate (e.g. using the same account on an iPhone and iPad), which [may be the result of a change on Apple's end.](https://twitter.com/_Matchstic/status/1194298114225721344)

I tried to switch my account by signing out and entering a different Apple ID, which appeared to succeed. However, ReProvision still functioned using the original credentials. Uninstalling and re-installing the .deb had no effect.

I had to do a DELETE FROM genp WHERE agrp like 'AAAAAAAAAA.com.matchstic.reprovision.ios' on my keychain-2.db to get ReProvision working with a different Apple ID.

**Comment by Haeckli** [comment]
I see the same. Can't sign my iPhone and iPad using the same AppleID. Already tried different IDs but always only allows signing 1 device at a time. For signing on the other device work, I have to remove the certificate first. 
Until the recent change, I always used just one AppleID for my 2 devices....

**Comment by amandamana** [comment]
@Haeckli I only mentioned the known 1-device issue you confirmed as background for this report of an issue switching to a different Apple ID. Have you tried using different Apple IDs on each of your devices or are you just removing/revoking certs manually?

**Comment by Matchstic** [comment]
Hm. Could be related to an issue I’ve noticed myself too when logging out - old credentials seem to be persisting for some reason.

**Comment by Matchstic** [comment]
As for the 1-device issue, I’m fairly sure that’s an Apple restriction - they must have narrowed down the device count a free certificate can be used on simultaneously. Not surprising really!

**Comment by amandamana** [comment]
> As for the 1-device issue, I’m fairly sure that’s an Apple restriction - they must have narrowed down the device count a free certificate can be used on simultaneously. Not surprising really!

I'd bet you're right. However, if ReProvision has some issue seeing/managing its keychain credentials it might be possible that some "ghost" certs are unknowingly contributing to the limit.

**Comment by Haeckli** [comment]
@amandamana I currently revoke the one certificate from the devices when resigning is due. But if its an apple-thing that they restrict free accounts to just 1 cert I'll probably open another free account...

**Comment by Matchstic** [comment]
Is this now fixed?

**Comment by DeafEyeJedi** [comment]
Worked for me. Running iOS 12.4 iPhone XR with latest ReProVision experimental update. Thanks, Matt!

**Comment by amandamana** [comment]
I updated two iOS 12.4/unc0ver devices to experimental6-1 and logged in with the two different Apple IDs used on each device before the update. I verified proper signing of unc0ver with the respective IDs. I then signed out on device #2 and logged back in using the same Apple ID as device #1. ReProvision functioned using the new credentials, great job! 
Also there seemed to be no issue using two certificates with my free account. I'm curious if these two issues were in fact related or just a coincidence.
